uxarray/grid.py

```diff
@@ -155,9 +209,10 @@ def __from_ds__(self, dataset):
         elif self.mesh_filetype == "ugrid":
             self.ds, self.ds_var_names = _read_ugrid(dataset, self.ds_var_names)
         elif self.mesh_filetype == "shp":
-            self.ds = _read_shpfile(self.filepath)
+            self.ds = _read_shpfile(dataset)
```

Comment:
This might need a revisit; maybe we should open an issue for this because:

- There won't be a shape file reading anymore (We will instead read in a dataset(?) that is already constructed from a shape file, how?)
- Our parse will ned to dete a shape file from such a dataset, not from a file extension, etc.